Source/QED/amrex_rng_wrapper.cpp

```diff
@@ -0,0 +1,20 @@
+//This file provides a wrapper aroud the RNG
```

Comment:
The QED library developed in PICSAR requires a "random number generator"
the only requirement is that it should provide
- a unf (a,b) function to generate uniformly distributed numbers between a and b
- an exp(l) function, to generate exponentially distributed numbers

Reply:
Actually, it seems to me that `amrex::RandomNormal` generates a random variable distributed according to a Gaussian distribution. I need exp(l) to generate numbers distributed according to an exponential distribution. It could be possible to do a pull request to implement  `amrex::RandomExp` in AMReX. However, care should be taken since curand does not implement the exponential distribution, as far as I know.

> Could you give it a more human-readable name, e.g., `RandomUniform` ?

Yes, but it will take time, since it is the name used in the QED module of the PICSAR library

>Does it have anything specific to the BW engine, or could it be used somewhere else? In the latter case, could you remove mentions to BW in this file, and move the file to Utils?

I'll try to make this point clearer. Sorry for not having done that before.
All the engines (Breit Wheeler, Quantum Synchrotron and Schwinger) in the QED module of PICSAR are template classes:
```
     template<typename _REAL, class _RNDWRAP>
     class breit_wheeler_engine

     template<typename _REAL, class _RNDWRAP>
     class quantum_synchrotron_engine

     template<typename _REAL, class _RNDWRAP>
     class schwinger_pair_engine
```

_RNDWRAP should be a class able to provide the 'unf' and 'exp' functions. 
Actually, this is only needed if the engines are instantiated and used as objects
to update the optical depth and to calculate the outcome of a QED process. 
At present, generating the lookup tables requires to instantiate the engine, but
the RNG is not actually invoked. Although the engine classes could be instantiated on 
a GPU, we agreed to use only calls to static functions in GPU kernels. These calls do
not invoke the RNG, since a random number is passed as an argument. 
In principle I could get rid of this class entirely, rely only on static memeber function
calls in GPU kernels and do something like the following to generate the lookup tables:

```
class rng_dummy{};

breit_wheeler_engine<amrex::Real, rng_dummy> bw_engine;

```

Do you think that this solution could be preferable? 
In any case, my impression is that this solution would benefit from an intermediate layer (e.g. a wrapper class) between the QED library and WarpX. Please see my comment specifically on this point.

Reply:
> Could you give it a more human-readable name, e.g., RandomUniform ?
> > Yes, but it will take time, since it is the name used in the QED module of the PICSAR library
Ok sorry I was confused about that. No need to change the names (in particular if we get rid of this class, as you suggest).

Ok thanks for clarifying that point. Yes, I think that if you won't need the random number generator, we can get rid of this class and pass a dummy RNG class to the initialization.

And in any case, yes, let's keep a wrapper layer to make future changes easier :)